api/dataset.go

```diff
@@ -2,38 +2,218 @@ package api
 
 import (
 	"context"
+	"errors"
 	"net/http"
+	"strings"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
+	"github.com/go-chi/chi/v5"
 )
 
-func (api *API) getDatasetJSON(w http.ResponseWriter, r *http.Request) {
-	var response *getDatasetJsonObservationsResponse
-	var err error
+type optionsMap map[string]map[string]dataset.Option
+
+type datasetParams struct {
+	id                 string
+	edition            string
+	version            string
+	basedOn            string
+	unsortedDimensions []string
+	options            optionsMap // dimension -> option -> option item
+}
 
+func (api *API) getDatasetJSON(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 
 	if r.URL.Query().Get("geography") == "" {
-		response, err = api.getDefaultDatasetJSON(ctx, r)
+		api.getDefaultDatasetJSON(ctx, w, r)
 	} else {
-		response, err = api.getGeographyDatsetJSON(ctx, r)
+		api.getGeographyDatsetJSON(ctx, r)
+	}
+}
+
+func (api *API) getDefaultDatasetJSON(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	params, err := api.getDatasetParams(ctx, r)
+
+	if err != nil {
+		api.respond.Error(ctx, w, http.StatusNotFound, err)
+		return
+	}
+
+	geoDimensions, err := api.getGeographyTypes(ctx, params.basedOn)
+	if err != nil {
+		api.respond.Error(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	dimensions, _ := api.sortGeography(geoDimensions, params.unsortedDimensions)
+
+	datasetRequest := cantabular.StaticDatasetQueryRequest{
+		Dataset:   params.basedOn,
+		Variables: dimensions,
+	}
+
+	queryResult, err := api.ctblr.StaticDatasetQuery(ctx, datasetRequest)
+
+	if err != nil {
+		api.respond.Error(ctx, w, http.StatusInternalServerError, err)
+		return
 	}
 
+	response, err := api.toGetDatasetJsonResponse(params.options, queryResult)
+
 	if err != nil {
-		api.respond.Error(
-			ctx,
-			w,
-			http.StatusBadRequest,
-			err,
-		)
+		api.respond.Error(ctx, w, http.StatusBadRequest, err)
 		return
 	}
 
 	api.respond.JSON(ctx, w, http.StatusOK, response)
 }
 
-func (api *API) getDefaultDatasetJSON(ctx context.Context, r *http.Request) (*getDatasetJsonObservationsResponse, error) {
+func (api *API) getGeographyDatsetJSON(ctx context.Context, r *http.Request) (*getDatasetJsonObservationsResponse, error) {
 	return &getDatasetJsonObservationsResponse{}, nil
 }
 
-func (api *API) getGeographyDatsetJSON(ctx context.Context, r *http.Request) (*getDatasetJsonObservationsResponse, error) {
-	return &getDatasetJsonObservationsResponse{}, nil
+func (api *API) toGetDatasetJsonResponse(dimensionMap optionsMap, query *cantabular.StaticDatasetQuery) (*getDatasetJsonObservationsResponse, error) {
+	var dimensions []getDatasetJsonResponseDimension
```

Comment:
Sorry I missed the initial PR but there seems to be a lot of extra structs with obtuse names that should be organised a lot better. A lot of structs are duplicates of what we already have. We have quite a particular way of keeping things organised and I would like to keep it that way, otherwise things will quickly become unbearable to work with. I will give some concrete examples below

Reply:
In contract.go:
- `getDatasetJsonResponseLink`, and `getDatasetJsonResponseDimensionOption` are unnecessary, please use the `model.Link` struct as we have elsewhere.
- I'm wondering why `getDatasetJsonResponseVersionLink` has a `version` field instead of an `id`? Everywhere else through the API versions are referred to purely by an HREF as some links are. If this is definitely correct then it would be better to keep the naming simpler i.e.`VersionLink`. We don't need each struct to only belong to one other struct, we want to re-use things as much as possible.
- `getDatasetJsonObservationsResponse` is not in line with our naming convention. If the endpoint/handler is called `getDatasetJSON` then the response should be called `getDatasetJSONResponse`. It should be a 1:1 match.
- For `getDatasetJsonResponseLinks` and `getDatasetJsonResponseDimension` you don't need to specify 'get' 
 and 'response'. `DatasetJSONLinks` and `DatasetJSONDimension` would be enough. Although `DatasetJSONDimensionItem` might be more in line with other naming in the package. Either way, we want to strive to keep type and variable names as short/simple as possible while retaining the useful information.